scripts/build/README.md

```diff
@@ -0,0 +1,58 @@
+# CHIP automated builds scripts
+
+CHIP compilation is generally split into two steps
+
+1. Generate ninja/makefile for out-of-source builds
+2. Compilation using ninja/makefiles
+
+## Building manually
+
+Manual building is generally platform-dependent. All build steps would require a
+bootstrapped environment (loads a pigweed build environment) and will then be
+followed by platform-specific instructions.
+
+The file BUILDING.md describes general requirements and examples. Typical usage
+is:
+
+```
+source scripts/activate
+gn gen out/host
+ninja -C out/host
+```
+
+## Unified build script
+
+The script `chipbuild.py` provides a single entry point for generating and
+executing the build.
+
+Build environment _MUST_ be properly configured for chipbuild to succeed. For
+example ESP32 builds requite IDF_PATH to be set. Building in the corresponding
+build image or the chip vscode image satisfy the build environment requirement.
+
+Usage examples:
+
+1. Compiles the Lock app on all supported platforms
+
+    ```
+    ./scripts/build/chipbuild.py --app lock build
+    ```
+
+2. Compile the all clusters app for a ESP32 DevKitC
+
+    ```
+    ./scripts/build/chipbuild.py --app all_clusters_app --board devkitc build
+    ```
+
+3. Generate all the makefiles (but do not compile) all native apps
+
+    ```
+    ./scripts/build/chipbuild.py --platform native generate --out-prefix out
```

Comment:
Is the out-prefix here required? If not, I'd remove it from the example....

We can have a separate example demonstrating how to use out-prefix.

Reply:
Moved as a separate example - 'out' is the default but can specify a different one.